src/com/google/enterprise/adaptor/testing/MockResponse.java

```diff
@@ -0,0 +1,221 @@
+// Copyright 2011 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.enterprise.adaptor.testing;
+
+import com.google.enterprise.adaptor.Acl;
+import com.google.enterprise.adaptor.Metadata;
+import com.google.enterprise.adaptor.Response;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A mock implementation of {@link Response} that simply caches the
+ * values it receives.
+ */
+/* TODO(jlacey): Implement Response2. */
+public class MockResponse implements Response {
+  private final OutputStream os;
+  private boolean notModified;
+  private boolean notFound;
+  private boolean noContent;
+  private String contentType;
+  private Date lastModified;
+  private Metadata metadata = new Metadata();
+  private Acl acl;
+  private Map<String, Acl> namedResources = new HashMap<String, Acl>();
```

Comment:
It's unnecessary. Set and Map equality are already order-independent, and equality is all the tests care about here, where they have the map itself to hand. On the other hand, if we were testing from other end of the wire (e.g., the GSA), then this map would get stringified, and we would need to care about the iteration order in order to (easily) check equality.